### DISCUSSION TITLE
trouble starting a new project with latest versions

### DISCUSSION THREAD

**Discussion by dtau00** [appy-one/acebase#103] [Q&A]
i have a project with everything working a couple of versions old (acebase-server 1.9.1).  I tried updating to the latest versions and immediatey ran into this error: 

`"...Cannot store undefined values. Set removeVoidProperties option to true to automatically remove undefined properties". `

So i created a brand new project, and again running into this:

```
E:\Projects\acebase-test\node_modules\acebase\src\storage-acebase.js:3614
                    throw new Error(`Property "${key}" has invalid value. Cannot store undefined values. Set removeVoidProperties option to true to automatically remove undefined properties`);      
                          ^

Error: Property "account" has invalid value. Cannot store undefined values. Set removeVoidProperties option to true to automatically remove undefined properties
    at E:\Projects\acebase-test\node_modules\acebase\src\storage-acebase.js:3614:27
    at Array.forEach (<anonymous>)
    at _writeNode (E:\Projects\acebase-test\node_modules\acebase\src\storage-acebase.js:3596:28)
    at _lockAndWriteNode (E:\Projects\acebase-test\node_modules\acebase\src\storage-acebase.js:3520:34)
    at async Promise.all (index 0)
    at async _writeNode (E:\Projects\acebase-test\node_modules\acebase\src\storage-acebase.js:3633:5)
    at async _lockAndWriteNode (E:\Projects\acebase-test\node_modules\acebase\src\storage-acebase.js:3520:28)
    at async Promise.all (index 0)
    at async _writeNode (E:\Projects\acebase-test\node_modules\acebase\src\storage-acebase.js:3633:5)
    at async _mergeNode (E:\Projects\acebase-test\node_modules\acebase\src\storage-acebase.js:3475:25)
```

tried adding removeVoidProperties to the options of the connection string to no avail.  Not sure what to try next.  Here's the test file:

```
const { AceBaseServer } = require('acebase-server');
const dbname = 'test_db'
const options = {   
    host: '0.0.0.0', port: 3005, 
    authentication: {
        enabled: true,
        allowUserSignup: true,
        defaultAccessRule: 'auth',
        defaultAdminPassword: 'testdbpw'
    },
    logLevel: 'log', 
    autoGrow: true,
    storage: { path: '.' } ,
    removeVoidProperties:true
};

const server = new AceBaseServer(dbname,options);
server.ready(async () => {
    console.log('ready')
});
```

**Comment by appy-one** [comment]
I am not able to reproduce this with your code sample?

**Reply by appy-one** [reply]
Let me know if this issue has been resolved or not

**Reply by dtau00** [reply]
not resolved, ill have to try again in the next week or so.  this might be related to some of the really odd issues i reported about a couple of weeks ago.  like something broke internally within a node of the db, and causing all types of ruckus.  this issue is odd since im starting over with a new project /db, ill probably have to do a deeper investigation

**Reply by appy-one** [reply]
I notice now that you are using `removeVoidProperties: true` in the wrong place, it is a `storage` setting so you should move it there: `storage: { path: '.', removeVoidProperties:true }`
